toolsrc/src/vcpkg/commands.list.cpp

```diff
@@ -4,12 +4,51 @@
 #include <vcpkg/commands.h>
 #include <vcpkg/help.h>
 #include <vcpkg/vcpkglib.h>
+#include <vcpkg/base/json.h>
 
 namespace vcpkg::Commands::List
 {
     static constexpr StringLiteral OPTION_FULLDESC =
         "--x-full-desc"; // TODO: This should find a better home, eventually
 
+    static constexpr StringLiteral OPTION_JSON = "--json";
+
+    static void do_print_json(std::vector<const vcpkg::StatusParagraph*> installed_packages)
+    {
+        Json::Object obj;
+
+        for (const StatusParagraph* status_paragraph : installed_packages)
+        {
+            auto current_spec = status_paragraph->package.spec;
+            if (!obj.contains(current_spec.to_string()))
+            {
+                Json::Object& library_obj = obj.insert(current_spec.to_string(), Json::Object());
+                library_obj.insert("package_name", Json::Value::string(current_spec.to_string()));
+                library_obj.insert("triplet", Json::Value::string(current_spec.triplet().to_string()));
+                library_obj.insert("version", Json::Value::string(status_paragraph->package.version));
+                Json::Array arr;
+                if (status_paragraph->package.is_feature())
+                {
+                    arr.push_back(Json::Value::string(status_paragraph->package.feature));
+                }
+                library_obj.insert("features", std::move(arr));
+                library_obj.insert("desc",
+                                   Json::Value::string(Strings::join("\n", status_paragraph->package.description)));
+            }
+            else
+            {
+                if (status_paragraph->package.is_feature())
+                {
+                    auto library_obj = obj.get(current_spec.to_string());
+                    auto& feature_list = library_obj->object().get("feature")->array();
```

Comment:
```suggestion
                    auto& feature_list = library_obj->object()["features"].array();
```

This'll fail on `"features"` not existing, as opposed to UB-ing.